pkg/microservice/aslan/core/common/repository/mongodb/ai/EnvAIAnalysis.go

```diff
@@ -0,0 +1,102 @@
+package ai
+
+import (
+	"context"
+	"errors"
+
+	"github.com/koderover/zadig/pkg/microservice/aslan/config"
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/common/repository/models/ai"
+	mongotool "github.com/koderover/zadig/pkg/tool/mongo"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type EnvAIAnalysisColl struct {
+	*mongo.Collection
+
+	coll string
+}
+
+func NewEnvAIAnalysisColl() *EnvAIAnalysisColl {
+	name := ai.EnvAIAnalysis{}.TableName()
+	return &EnvAIAnalysisColl{Collection: mongotool.Database(config.MongoDatabase()).Collection(name), coll: name}
+}
+
+func (c *EnvAIAnalysisColl) GetCollectionName() string {
+	return c.coll
+}
+
+func (c *EnvAIAnalysisColl) EnsureIndex(ctx context.Context) error {
+	mod := []mongo.IndexModel{
+		{
+			Keys: bson.D{
+				bson.E{Key: "project_name", Value: 1},
+				bson.E{Key: "env_name", Value: 1},
+				bson.E{Key: "production", Value: 1},
```

Comment:
production is a bool value, it's not efficiency when use a bool as an index.

And the env_name can't be same in the same project, so it's not necessary here.